rs/consensus/src/consensus/block_maker.rs

```diff
@@ -514,17 +516,51 @@ pub(crate) fn already_proposed(pool: &PoolReader<'_>, h: Height, this_node: Node
         .any(|p| p.signature.signer == this_node)
 }
 
+const DYNAMIC_DELAY_LOOK_BACK_DISTANCE: Height = Height::new(29);
+const DYNAMIC_DELAY_MAX_NON_RANK_0_BLOCKS: usize = 5;
+const DYNAMIC_DELAY_EXTRA_DURATION: Duration = Duration::from_secs(10);
+
+fn count_non_rank_0_blocks(pool: &PoolReader, block: Block) -> usize {
```

Comment:
Would it make sense to add some documentation describing why this logic is needed and why these values were chosen?

Reply:
Done